spec/solver_spec.rb

```diff
@@ -0,0 +1,49 @@
+require_relative 'spec_helper'
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+  describe 'factorial' do
+    it 'Create new instance of solver class' do
+      @solver.should be_an_instance_of Solver
+    end
+
+    it 'Return the factorial of the number without breaking' do
+      expect(@solver.factorial(4)).to eql 24
+    end
+
+    it 'Return 1 if the input is 0' do
+      expect(@solver.factorial(0)).to eql 1
+    end
+
+    it 'Raises an exception' do
+      # expect(@solver.factorial(-9)).to eql
```

Comment:
- [ ] [OPTIONAL] Kindly remove any unused commented code from your repo, it makes your code messy